vm/systemSmartContracts/auction.go

```diff
@@ -1424,70 +1419,18 @@ func (s *stakingAuctionSC) unBondNecessaryNodes(
 	return totalUnBond, unBondedNodes, nil
 }
 
-func (s *stakingAuctionSC) unBondTokensWithNodes(args *vmcommon.ContractCallInput) vmcommon.ReturnCode {
-	registrationData, returnCode := s.basicCheckForUnStakeUnBond(args)
-	if returnCode != vmcommon.Ok {
-		return returnCode
-	}
-	err := s.eei.UseGas(s.gasCost.MetaChainSystemSCsCost.UnBondTokens)
-	if err != nil {
-		s.eei.AddReturnMessage(vm.InsufficientGasLimit)
-		return vmcommon.OutOfGas
-	}
-	if len(args.Arguments) != 1 {
-		s.eei.AddReturnMessage("invalid num of arguments")
-		return vmcommon.UserError
-	}
-
-	unBondValue := big.NewInt(0).SetBytes(args.Arguments[0])
-	if unBondValue.Cmp(zero) <= 0 {
-		s.eei.AddReturnMessage("cannot unBond negative value")
-		return vmcommon.UserError
-	}
-
-	totalUnBond, index := s.computeUnBondTokens(registrationData)
-	if totalUnBond.Cmp(zero) > 0 {
-		registrationData.UnstakedInfo = registrationData.UnstakedInfo[index:]
-		registrationData.TotalUnstaked.Sub(registrationData.TotalUnstaked, totalUnBond)
-		if registrationData.TotalUnstaked.Cmp(zero) < 0 {
-			s.eei.AddReturnMessage("contract error on unBondTokens function, total unstaked < total unbond")
-			return vmcommon.UserError
-		}
-	}
-
-	unBondedNodes := make([][]byte, 0)
-	totalUnBondedFromNodes := big.NewInt(0)
-	if totalUnBond.Cmp(unBondValue) < 0 {
-		remainingToUnBond := big.NewInt(0).Sub(unBondValue, totalUnBond)
-		totalUnBondedFromNodes, unBondedNodes, err = s.unBondNecessaryNodes(registrationData, remainingToUnBond)
-		if err != nil {
-			s.eei.AddReturnMessage(err.Error())
-			return vmcommon.UserError
-		}
-	}
-
-	totalUnBond.Add(totalUnBond, totalUnBondedFromNodes)
-
-	s.eei.Finish(totalUnBond.Bytes())
-	err = s.eei.Transfer(args.CallerAddr, args.RecipientAddr, totalUnBond, nil, 0)
-	if err != nil {
-		s.eei.AddReturnMessage("transfer error on unBond function")
-		return vmcommon.UserError
-	}
-
-	returnCode = s.updateRegistrationDataAfterUnBond(registrationData, totalUnBond, unBondedNodes, args.CallerAddr)
-	if returnCode != vmcommon.Ok {
-		return returnCode
-	}
-
-	return vmcommon.Ok
-}
-
-func (s *stakingAuctionSC) computeUnBondTokens(registrationData *AuctionDataV2) (*big.Int, int) {
+func (s *stakingAuctionSC) unBondTokensFromRegistrationData(
```

Comment:
Add some unit tests for this func.

Reply:
already added.

Reply:
there are several which test unBond - which calls this func internally